examples/deadletterqueue/movingMessagesToDLQ.ts

```diff
@@ -0,0 +1,192 @@
+
+    This sample demonstrates usage of DLQ for 3 such scenarios:
+      A. When client invokes the .deadletter() API on the brokered message.
+      B. When system exceeds maximum allowed delivery attempts on the message,
+         which is by default set to 10
+      C. When processing of message encounters error or an exception.
+
+    After running this sample, run the processMessagesInDLQ example to see how the messages
+    can be reprocessed.
+
+*/
+async function main(): Promise<void> {
+  try {
+    ns = Namespace.createFromConnectionString(str);
+    // Clean up queues before running.
+    await purgeAllQueues();
+
+    // This scenario demonstrates brokered message's deadletter() API resulting in immediate
+    // message move to DLQ.
+    await runSimpleDeadLetterScenario();
+
+    // This scenario demonstrates how exceeding delivery count on message will result in
+    // automatic move to DLQ
+    await runExceedMaxRetriesScenario();
+
+    // This scenario demonstrates how message processing failures can cause message move to DLQ.
+    await runProcessingErrorHandlingScenario();
+  } catch (err) {
+    console.log(">>>>> Error in running sample scenarios: ", err);
+  } finally {
+    console.log("\n >>>> Calling close....");
+    ns.close();
+  }
+}
+
+// Queue clean up util
+async function purgeAllQueues(): Promise<void> {
+  await purgeQueue(queuePath);
+  await purgeQueue(queuePath + "/$DeadLetterQueue");
+}
+
+async function purgeQueue(inputQueuePath: string): Promise<void> {
+  // Process messages from queue, by invoking .deadletter() on the brokered message
+  await processMessageQueue(queuePurgingProcessor, inputQueuePath);
+  console.log(">>>> Cleaning up queue: ", inputQueuePath);
+}
+
+async function runSimpleDeadLetterScenario(): Promise<void> {
+  // Prepare given queue by sending few messages
+  await sendMessages();
+
+  // Process messages from queue, by invoking .deadletter() on the brokered message
+  console.log("\n Setup queues, now running scenario A ... ");
+  await processMessageQueue(simpleDeadletterProcessor, queuePath);
+}
+
+async function runExceedMaxRetriesScenario(): Promise<void> {
+  // Prepare given queue by sending few messages
+  await sendMessages();
+
+  // Process messages from queue - exceed max retries on the messages
+  console.log("\n Setup queues, now running scenario B ... ");
+  await processMessageQueue(exceedMaxRetriesProcessor, queuePath);
+}
+
+async function runProcessingErrorHandlingScenario(): Promise<void> {
+  // Prepare given queue by sending few messages
+  await sendMessages();
+
+  // Process messages from queue, by invoking a faulty processor that doesn't successfully process
+  console.log("\n Setup queues, now running scenario C ... ");
+  await processMessageQueue(faultyProcessor, queuePath);
+}
+
+// Helper to send few sample messages to given queue
+async function sendMessages(): Promise<void> {
```

Comment:
We have a similar method in other samples (that one sets the TTL on the message as well as make the messages be distinct). I suggest you take a look at that method and see if we can re-use it here.